src/services/mail.service.js

```diff
@@ -0,0 +1,36 @@
+import nodemailer from 'nodemailer';
+
+const transporter = nodemailer.createTransport({
+  host: process.env.SMTP_HOST,
+  port: process.env.SMTP_PORT,
+  auth: {
+    user: process.env.SMTP_USER,
+    pass: process.env.SMTP_PASSWORD,
+  },
+});
+
+export const sendEmail = ({ to, subject, text, html }) => {
+  return transporter.sendMail({
+    from: `Node Auth App <${process.env.SMTP_FROM}>`,
+    to,
+    subject,
+    text: text ?? '',
+    html,
+  });
+};
+
+export const sendActivationEmail = (to, token) => {
+  const link = `http://${process.env.API_HOST}:${process.env.API_PORT}/activate/${token}`;
```

Comment:
It's better to use HTTPS instead of HTTP for the activation link to ensure secure communication. Consider updating the protocol to HTTPS if your server supports it.